File: Tree/Traverals/C++/postOrder.cpp
#include <iostream>
#include <queue>
using namespace std;

struct Node{
    //This is the structure of the binary tree.
    int data;
    Node *left;
    Node *right;
};

void postOrder(Node *root)
{
	if(root!=NULL)
	{
		postOrder(root->left);
		postOrder(root->right);
        cout << root->data << " ";
	}
}


int main(int argc, char const *argv[])
{
    // Initializing a 3 element binary tree.
   Node *root = new Node();
    root->data = 10;
    root->right = new Node();
    root->right->data = 15;
    root->left = new Node();
    root->left->data = 5;
    root->left->left = new Node();
    root->left->left->data = 30;
    root->left->right = new Node();
    root->left->right->data = 50;;

    postOrder(root);
    return 0;
}    

File: Tree/Traverals/JAVA/postorder.java
import java.io.*;
import java.util.*;
  
  class postorder{
 private Node root;
   private class Node{

    private Node left;
    private Node right;
   
    int data;
   
    public Node(int data){
       this.data=data;  
       System.out.print(data+"\t");       
    }
  }

  public void createBinaryTree(){


    Node first = new Node(1);
    Node second = new Node(2);
    Node third = new Node(3);
    Node fourth = new Node(4);
    Node fifth = new Node(5);
   
     root = first;
     first.left = second;
     first.right = third;
     second.left = fourth;
     second.right = fifth;
  }

   public void postorders(Node root){

      if(root == null){
      return ; 
     }
     postorders(root.left);
     postorders(root.right);
     System.out.print(root.data+"\t");
   } 



 public static void main(String args[]){

   postorder obj = new postorder();
   System.out.println("Before post");
   obj.createBinaryTree();
   System.out.println();
   System.out.println("After postorder");
   obj.postorders(obj.root);
 }
}
